src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -475,44 +474,53 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     }
     const duration = this.props.graphDuration.value;
     // Keep a map of namespace x promises in order not to fetch several times the same data per namespace
-    const healthPerNamespace = new Map<String, Promise<NamespaceHealth>>();
+    const appHealthPerNamespace = new Map<String, Promise<NamespaceAppHealth>>();
+    const wkldHealthPerNamespace = new Map<String, Promise<NamespaceWorkloadHealth>>();
     // Asynchronously fetch health
     cy.nodes().forEach(ele => {
       const namespace = ele.data('namespace');
-      const app = ele.data('app');
-      const workload = ele.data('workload');
-
-      if ((workload !== 'unknown' || app !== 'unknown') && (ele.data('isGroup') || !ele.data('parent'))) {
-        const appBasedHealth = this.props.graphType === GraphType.APP && app !== 'unknown';
-        let promise = healthPerNamespace.get(namespace);
-        if (!promise) {
-          promise = API.getNamespaceHealth(authentication(), namespace, duration).then(r => r.data);
-          healthPerNamespace.set(namespace, promise);
+      if (ele.data('isWorkload')) {
```

Comment:
I suppose this only appears (isWorkload) when the graphType === workload or when there is no "app" label (regardless of graphType). is that right?

Reply:
@josejulio Yes, I believe that is true.

Note that this "isWorkload" attribute is soon going away. @jshaughn is refactoring this - "isWorkload" will soon be replaced with "nodeType" attribute whose value will be something like "app", "workload", or "service". @jshaughn can give specifics, but that will be the basic idea.

Reply:
That's great. I'd need one confirmation about it:
In app-versioned graph, a node that is inside a group will be of type "workload", right? And a node that is not in group is of type "app". Correct? (cc @jshaughn )

Reply:
@jotak , @jmazzitelli , no I don't think that is correct.   For a versionedApp graph the nodes inside a group should be NodeTypeApp where  n.app and n.version are both set.  The n.app would be the name on the group and the n.version would be the name displayed for the node.  The only time a node would be NodeTypeWorkload in an app graph is if the workload is not properly labeled and so we can't set the app/version fields.

But, for a versionedApp graph the nodes of type NodeTypeApp will still have n.workload set.  This is because, unless misconfigured, the unique app+version should equate to a unique workload.  That workload name can then be used for metric gathering and health, as opposed to the app+version labels.

Reply:
Ok so it won't help me, I'll have to go back to my first idea which was to detect if the node is in group or not.

Reply:
I guess I don't understand why you would expect it to be a workload, the groups are specifically for boxing app versions.  But I also don't know why it doesn't help you, the workload field is set on the node.  Also, if you need to know if a node is in a group you can look for ele.data('parent').  We should talk this out because my intent was to support your work as best as possible.

Reply:
> Ok so it won't help me, I'll have to go back to my first idea which was to detect if the node is in group or not.

I think you can use `ele.data('workload')`

Reply:
Let's talk about it on monday :)